cpp/RNSScreenRemovalListener.h

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <react/renderer/componentregistry/ComponentDescriptorFactory.h>
+#include <react/renderer/mounting/MountingOverrideDelegate.h>
+#include <react/renderer/mounting/ShadowView.h>
+
+using namespace facebook::react;
+
+struct RNSScreenRemovalListener : public MountingOverrideDelegate {
+  std::function<void(int)> listenerFunction_;
+  RNSScreenRemovalListener(std::function<void(int)> &&listenerFunction_)
+      : listenerFunction_(listenerFunction_) {}
```

Comment:
haven't opened the code, writing from memory: don't we need here to have `std::move` / `std::forward` called to invoke move constructor of `std::function`?

```suggestion
      : listenerFunction_(std::move(listenerFunction_)) {}
```

Reply:
It works so I guess we don't?

Reply:
It is not matter of whether it works, it is matter of how it works. 

See [this snippet](https://onlinegdb.com/Sc6hCE0Lk) for the explanation.

Heres the example code in case the snippet times out (you can paste it to an online C++ compiler, e.g. [this one](https://www.onlinegdb.com/)

<details><summary>Code snippet</summary>
<p>

```c++
#include <iostream>

class SomeClass {
    int someField;

public:
    // DefaultConstructor
    SomeClass() : someField{0} {
        std::cout << "DefaultConstructor invoked\n";
    }

    // CopyConstructor
    SomeClass(const SomeClass& other) {
        std::cout << "CopyConstructor invoked\n";
        this->someField = other.someField;
    }

    // MoveConstructor
    SomeClass(SomeClass&& other) {
        std::cout << "MoveConstructor invoked\n";
        this->someField = std::move(other.someField);
    }
};

class CompositeClass {
    SomeClass someClassInstance;
public:
    CompositeClass(SomeClass&& other) : someClassInstance(other) {}

    // LOOK HERE!!!!
    // Comment the line above and uncomment the line below to see the difference.
    // CompositeClass(SomeClass&& other) : someClassInstance(std::move(other)) {}
};

// Type your code here, or load an example.
int main() {
    SomeClass someClassInstance{};
    CompositeClass ccInstance(std::move(someClassInstance));
    return 0;
}
```

</p>
</details>

Reply:
And now, after testing I'm sure we shall have `std::move` there.

Reply:
Can you explain what is the difference exactly and how it affects the code I added?

Reply:
I provided a code snippet with mimicking the setup you have here in this PR & appended comments explaining. 
The difference comes down to which constructor is being called & that's why I've put "couts" in ctors of `SomeClass` class.

Reply:
Basically right now you take rvalue reference as `RNSScreenRemovalListener` constructor parameter just to later pass it *by value* to initialiser of `listenerFunction_` field effectively copying it (invoking copy constructor). By moving there, as I've suggested, you invoke move constructor of `std::function` forwarding the parameter value correctly w/o unnecessary copy.

Reply:
Done